packages/server-side-render/src/server-side-render.js

```diff
@@ -140,7 +146,7 @@ export default function ServerSideRender( props ) {
 			} ) );
 
 		return fetchRequest;
-	}
+	}, [ isMountedRef, prevProps ] );
```

Comment:
Just a side note that since we no longer use Lodash in the entire repository (see #52571), we're using a couple of methods for comparison:

* shallow comparison: see the `@wordpress/is-shallow-equal` package.
* deep comparison: the `fast-deep-equal` package.